File: BMI/app/src/main/java/com/pjatk/s16281/MainActivity.java
package com.pjatk.s16281;

import android.os.Bundle;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.snackbar.Snackbar;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.view.View;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private EditText weight;
    private EditText height;
    private TextView result;
    private Button calc;
    private double bmi;
    private String answer;
    private Exception error;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);


        FloatingActionButton fab = findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Snackbar.make(view, "I tak jesteś gruby", Snackbar.LENGTH_LONG)
                        .setAction("Action", null).show();
            }
        });
        // Remake
        weight = findViewById(R.id.weight);
        height = findViewById(R.id.height);
        result = findViewById(R.id.result);
        calc = findViewById(R.id.calculate);

        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Snackbar.make(view, "You are fat anyway", Snackbar.LENGTH_LONG)
                        .setAction("Action", null).show();
            }
        });

        calc.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                calculateBmi();
            }
        });
        //

    }

    public void calculateBmi(){
        String inputWeightTxt = weight.getText().toString();
        String inputHeightTxt = height.getText().toString();

        double weight = Double.parseDouble(inputWeightTxt);
        double height = Double.parseDouble(inputHeightTxt);

        try{
            bmi = weight / Math.pow(height/100, 2);
            displayBmi();
        }
        catch (ArithmeticException ex){
            error = ex;
        }

        if (height == 0){
            result.setText("Height cannot be 0");
        }

    }

    private void displayBmi(){
        if ( bmi < 16 )
            answer = bmi+": wygłodzenie";
        else if(bmi>=16 && bmi<17)
            answer = bmi+": wychudzenie";
        else if(bmi>=17 && bmi<18.5)
            answer = bmi+": niedowaga";
        else if(bmi>=18.5 && bmi<24.5)
            answer = bmi+": pożądana masa ciała";
        else if(bmi>=24.5 && bmi<30)
            answer = bmi+": nadwaga";
        else if(bmi>=30 && bmi<35)
            answer = bmi+": otyłość I stopnia";
        else if(bmi>=35 && bmi<40)
            answer = bmi+": otyłość II stopnia (duża)";
        else if(bmi>=40)
            answer = bmi+": otyłość III stopnia (chorobliwa)";
        else
            answer = "Error! You can't be so fat";

        result.setText("BMI: "+ answer);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: KOTLIN/app/src/main/java/com/pjatk/lab4/MainActivity.kt
package com.pjatk.lab4

import android.os.Bundle
import android.util.Log
import com.google.android.material.snackbar.Snackbar
import androidx.appcompat.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem
import androidx.fragment.app.FragmentManager
import androidx.navigation.findNavController
import androidx.navigation.fragment.FragmentNavigator
import androidx.navigation.fragment.findNavController

import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)

        fab.setOnClickListener { view ->
            Snackbar.make(view, getString(R.string.snackbar_main_message), Snackbar.LENGTH_LONG)
                    .setAction("Action", null).show()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.

        return when (item.itemId) {

            R.id.menu_main -> {
                Log.i("tag", "menu clicked")
                return true
            }
            R.id.menu_bmi -> {
                Log.i("tag","bmi clicked")
                return true
            }
            R.id.menu_ppm -> {
                Log.i("tag","ppm clicked")
                return true
            }
            R.id.menu_graph -> {
                Log.i("tag","graph clicked")
                return true
            }
            R.id.action_settings -> {
                Log.i("tag","settings clicked")
                return true
            }
            else -> super.onOptionsItemSelected(item)
        }




    }
}
